py/desispec/coaddition.py

```diff
@@ -490,16 +490,18 @@ def coadd_cameras(spectra, cosmics_nsig=0., onetile=False) :
             wave=np.append(wave,spectra.wave[b][spectra.wave[b]>wave[-1]+tolerance])
     nwave=wave.size
 
-    # check alignment
+    # check alignment, caching band wavelength grid indices as we go
+    windict = {}
     number_of_overlapping_cameras=np.zeros(nwave)
     for b in spectra.bands :
-        windices=np.argmin((np.tile(wave,(spectra.wave[b].size,1))-np.tile(spectra.wave[b],(wave.size,1)).T)**2,axis=1)
-        dist=np.sqrt(np.max(spectra.wave[b] - wave[windices]))
-        log.debug("camera {} max dist= {}A".format(b,dist))
-        if dist > tolerance :
-            log.error("Cannot directly coadd the camera spectra because wavelength are not aligned, use --lin-step or --log10-step to resample to a common grid")
+        windices = np.argwhere((np.abs(spectra.wave[b][:,None] - wave) < tolerance).any(0)).ravel()
```

Comment:
This still generates a large temporary 2D array in memory and I suspect that we could still get much much faster with a series of 1D operations, e.g. something like:
```
imin = np.argmin(np.abs(spectra.wave[b][0] - wave))
windices = np.arange(imin, imin+len(spectra.wave[b]), dtype=int)
dwave = spectra.wave[b] - wave[windices]
if np.any(np.abs(dwave) > tol):
    raise ValueError("Input wavelength grids aren't aligned...")
```
But that would need to be profiled to see if it actually is faster (and debugged to make sure it is correct).

Reply:
True, it's not memory efficient and we may be getting hurt even with a vectorized loop over the 2D structure. I'll mess around to see if there are more gains to be had.